### ISSUE TITLE
Advtrains - Update Station assignment?

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#255]
_Hedgehog ((Not quite certain what the exact idea or suggestion here is. Text goes all directions. If someone can clarify this, that would be great.  ))_

baijin
advtrains mod uses station assignment to display the route stops and endpoint with estimated time info on the player's HUD, and the speed of the train
Constant calculations aren't done, save first recorded info. also run a CSM client-side so the client can accurately predict and follow train route without throwing me out and keeping me on the tracks like it just did
Only constant speed along track vector, and accelerations around stations are needed

**Comment by yw05** [comment]
This sounds more like a request that should be sent to the advtrains mailing list, but here are my two cents on the matter:

About the HUD:
- The HUD shows the _distance_ to the nearest LZB point (temporary speed restriction point), not the time.
- Station and HUD displays are done separately.

I am not particularly sure about the context, but "[running] a CSM client-side so that client can accurately predict and follow train route" and "[not] throwing me out and keeping me on the tracks" sound like accurate route calculation and proper entity attachments (respectively) to me.

I do not have much (if at all) knowledge about MT's entity attachment mechanism, but the problem with train movements is that MT does not allow sending a (potentially curved/bent) path of an entity to the client, which I think is solved by the proposed CSM.

I would suggest writing a separate (server) mod for that, at least for now, as MT's modchannel does not (yet) allow sending messages to only one client. This can potentially be an issue on other servers where the position of other players and trains is not easily known.

I also don't think it is "only constant speed along track vector and accelerations around stations" that is needed, considering the speed restriction signs (etc) and advtrains' (milder) dtime clipping system that slows trains down. This will possibly become obvious when the new-ks branch gets pulled here.

Basically I would say that it is up to the engine to allow passing paths and advtrains to then support such path passing and a better movement logic ([which is, unfortunately, not likely going to be implemented unless there is a real benefit in doing so, as orwell wrote](https://lists.sr.ht/~gpcf/advtrains/%3C20211229152618.24873-1-tanavit%40laposte.net%3E#%3Cb8d5fbc1-231a-af39-a21a-335019b6dc94@bleipb.de%3E))
